### PR TITLE
[FIX] base_automation: avoid access right issues filter domains

### PR DESCRIPTION
**Pull request by nseinlet**
If some filter domains use M2O to models current user cannot access, 
using sudo(1) permit to filter even when user cannot access linked 
models.

### CODE REVIEW

**Comment by nseinlet** [comment]
@rco-odoo I don't think this is clean and I suspect this to hide the real issue

File: addons/base_automation/models/base_automation.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def _filter_pre(self, records):
         """ Filter the records that satisfy the precondition of action ``self``. """
         if self.filter_pre_domain and records:
             domain = [('id', 'in', records.ids)] + safe_eval(self.filter_pre_domain, self._get_eval_context())
-            return records.search(domain)
+            return records.sudo(1).search(domain).with_user(self.env.user.id)
```
Comment (MiquelRForgeFlow) [inline addons/base_automation/models/base_automation.py]:
```suggestion
            return records.sudo(SUPERUSER_ID).search(domain).with_user(self.env.user.id)
```
the same for the other

Comment (rco-odoo) [reply addons/base_automation/models/base_automation.py]: Don't change the user id!
```suggestion
            return records.sudo().search(domain).with_env(records.env)
```

File: addons/base_automation/models/base_automation.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def _filter_post_export_domain(self, records):
         """ Filter the records that satisfy the postcondition of action ``self``. """
         if self.filter_domain and records:
             domain = [('id', 'in', records.ids)] + safe_eval(self.filter_domain, self._get_eval_context())
-            return records.search(domain), domain
+            return records.sudo(1).search(domain).with_user(self.env.user.id), domain
```
Comment (rco-odoo) [inline addons/base_automation/models/base_automation.py]:
```suggestion
            return records.sudo().search(domain).with_env(records.env), domain
```

**Comment by rco-odoo** [comment]
@nseinlet in Odoo 13.0, `sudo()` activates a "superuser mode" that bypasses access rights without changing the current user.

**Comment by nseinlet** [comment]
@rco-odoo Added a test to reproduce the issue

File: addons/base_automation/models/base_automation.py [hunk 3]
```diff
@@ -156,7 +156,7 @@ def _filter_pre(self, records):
         """ Filter the records that satisfy the precondition of action ``self``. """
         if self.filter_pre_domain and records:
             domain = [('id', 'in', records.ids)] + safe_eval(self.filter_pre_domain, self._get_eval_context())
-            return records.search(domain)
+            return records.sudo().search(domain).with_user(self.env.user.id)
```
Comment (rco-odoo) [inline addons/base_automation/models/base_automation.py]: You have to put the result back with the original environment:
```suggestion
            return records.sudo().search(domain).with_env(records.env)
```
If the original environment already had the flag `su`, the call to `with_user` will reset that flag.

File: addons/base_automation/models/base_automation.py [hunk 4]
```diff
@@ -167,7 +167,7 @@ def _filter_post_export_domain(self, records):
         """ Filter the records that satisfy the postcondition of action ``self``. """
         if self.filter_domain and records:
             domain = [('id', 'in', records.ids)] + safe_eval(self.filter_domain, self._get_eval_context())
-            return records.search(domain), domain
+            return records.sudo().search(domain).with_user(self.env.user.id), domain
```
Comment (rco-odoo) [inline addons/base_automation/models/base_automation.py]: Same as above.

File: addons/base_automation/tests/test_base_automation.py [hunk 5]
```diff
@@ -205,3 +205,54 @@ def test_21_trigger_fields(self):
...
+            'active': True,
+            'code': "action = [rec.name for rec in records]"
+        })
+
+        self.env['base.automation.linked.test'].create([
+            {'name': 'a first record', 'another_field': 'something'},
+            {'name': 'another record', 'another_field': 'something different'},
+        ])
+
+        rec1_id = self.env['base.automation.link.test'].create({'name': 'a record'})
```
Comment (rco-odoo) [inline addons/base_automation/tests/test_base_automation.py]: Inconsistent naming convention: `rec1_id` suggests that the value is an id.  Use `rec1` instead.

**Comment by tfr-odoo** [comment]
@rco-odoo the branch has been updated

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**